Source/DafnyRuntime/DafnyRuntime.go

```diff
@@ -692,183 +692,412 @@ func (seq Seq) VerbatimString(asLiteral bool) string {
  * Arrays
  ******************************************************************************/
 
-// An Array is a Go slice representing a (possibly) multidimensional array,
-// along with metadata.  There aren't any methods for updating; instead, you can
-// update through the pointer returned by Index.
-type Array struct {
-  contents []interface{} // stored as a flat one-dimensional slice
-  dims     []int
-}
-
-func newArray(dims ...Int) *Array {
+// Array is the general interface for arrays. Conceptually, it contains some
+// underlying storage (a slice) of elements, together with a record of the length
+// of each dimension. Thus, this general interface supports 1- and multi-dimensional
+// Dafny arrays.
+//
+// All array indices are given as the Go type "int". This is what the Dafny implementation
```

Comment:
I'm only saying that the logic in the current Dafny-to-Go compiler selects `uint32` as the native type for the `newtype` declaration I gave above (and your PR doesn't affect that). It's not a problem for this PR since you're explicitly picking `int` for the array index type, and that does in fact line up with the type of the built-in `len()` function, for example.

I'm only asking about this because in MY PR, in order to replace some of the Go runtime with cross-compiled Dafny code, I will end up implicitly using `uint32` as the array index type, or more accurately as the sequence indexing type. I'm mentioning this in case you know of a reason that's a bad idea so I can adjust my approach. It's in no way an issue with THIS PR. Sorry for not being clearer. :)

Reply:
It seems alright to use something slightly different for the sequences. If the sequence implementation supports lazy concatenation, then it may actually support sequence lengths that exceed the maximum array length.